### DISCUSSION TITLE
Let's Generate a List of Missing Math AutoCorrect Codes for Commonly Used Symbols

### DISCUSSION THREAD

**Discussion by brichwin** [daisy/math-a11y#26] [General]
This is slightly different from the [Let's Generate a List of Aliases for Existing Math AutoCorrect Codes discussion](https://github.com/daisy/math-a11y/discussions/25) in that it is a place to discuss symbols that do not have existing Math AutoCorrect codes.

For example, the right angle symbol (ASCII 221F: ∟), the large circle (ASCII 22EF: ◯), the triangle (ASCII 25B3: △), etc.

I created a wiki page to make the list at: [Proposed New Math AutoCorrect Codes for Commonly Used Symbols](https://github.com/daisy/math-a11y/wiki/Proposed-New-Math-AutoCorrect-Codes-for-Commonly-Used-Symbols)

**Comment by jenmarsala** [comment]
Below are some symbols that are missing autocorrect codes. (Note that some of the symbols, such as "not less than" render properly in Firefox but not in Edge or Chrome.)

**Symbols that are missing autocorrect codes**

Name | Symbol | Keystrokes to produce the symbol | Recommended autocorrect code(s) | Notes
-- | -- | -- | -- | --
Circle | ◯ | 25ef Alt + x | \circle | The circle symbol at the left is larger than the letter    O, but could possibly be confused with the letter O. Note that the command \odot    produces a circle symbol with a dot inside, &#x2299;, which is read as “circle dot operator.”
Line | e.g., AB with &#x2194; above  | Produced using (AB)\tvec\<sp>\<sp>, or using the Accent category of the equation toolbar. | \line | Kudos to Brian for the tip on using the \tvec command! <br>Note that \overbar and \overline instead produce a segment symbol above the letters without arrows on each end, like &#x203E; .
Measured angle | &#x2221;  | 2221 Alt + x | \measangle |  
Right angle | ⦜ | 299c Alt + x | \rightangle    \90angle | This symbol does not seem to appear in any of the galleries of Word’s equation toolbar.
Triangle | △ | 25b3 Alt + x | \triangle | This is read as “white up-pointing triangle.”
Not less than | &#x226E;  | 226e Alt + x | \notless | The symbol is not rendering properly in Edge or Chrome (OK in Firefox).
Not greater than | &#x226F;  | 226f Alt + x | \notgreater | The symbol is not rendering properly in Edge or Chrome (OK in Firefox).
Neither less than nor equal to | &#x2270;  | 2270 Alt + x | \notleq<br>    \notle | The symbol is not rendering properly in Edge or Chrome (OK in Firefox).
Neither greater than nor equal to | &#x2271;  | 2271 Alt + x | \notgeq<br>    \notge | The symbol is not rendering properly in Edge or Chrome (OK in Firefox).
Not parallel | &#x2226;  | 2226 Alt + x | \notparallel |  
Not congruent to |  &#x2247; | 2247 Alt + x | \notcong |  
Not almost equal to | &#x2249;  | 2249 Alt + x | \notapprox | The symbol is not rendering properly in Edge or Chrome (OK in Firefox).
Not a subset of | &#x2284;  | 2284 Alt + x | \notsubset |  
Not a superset of | &#x2285;  | 2285 Alt + x | \notsuperset |  
Neither a subset of nor equal to | &#x2288;  | 2288 Alt + x | \notsubseteq |  
Neither a superset of nor equal to | &#x2289;  | 2289 Alt + x | \notsuperseteq |  
Subset of with not equal to | &#x228a;  | 228a Alt + x | \subsetnoteq |  
Superset of with not equal to | &#x228b;  | 228b Alt + x | \supersetnoteq |  
Does not divide | &#x2224;  | 2224 Alt + x | \notdivide |

**Reply by brichwin** [reply]
I was able to create line AB by entering `(AB)\tvec<sp><sp>` where the <sp> represents a space. In other words, type `(AB)\tvec` followed by two spaces. The first space builds up the autocorrect code into the Unicode combing character and the second space actually combines them. The parenthesis are needed so the double left-right arrow covers both the character A and the character B.

# Combining Characters Examples

The following are examples of Unicode combining characters applied to the letter "a" (Note, this renders nicely in Firefox. I could not get it to render in Edge or Chrome):

- \acute (U+0301): `a\u0301` → á (combining acute accent)
- \Bar (U+033F): `a\u033F` → a̿ (combining double overline)
- \bar (U+0305): `a\u0305` → a̅ (combining overline)
- \breve (U+0306): `a\u0306` → ă (combining breve above)
- \check (U+030C): `a\u030C` → ǎ (combining caron (wedge) above)
- \ddddot (U+20DC): `a\u20DC` → a⃜ (combining four dots above)
- \dddot (U+20DB): `a\u20DB` → a⃛ (combining three dots above)
- \ddot (U+0308): `a\u0308` → ä (combining double dot above, umlaut accent)
- \dot (U+0307): `a\u0307` → ȧ (combining dot above)
- \grave (U+0300): `a\u0300` → à (combining grave accent)
- \hat (U+0302): `a\u0302` → â (combining circumflex (caret) accent)
- \hvec (U+20D1): `a\u20D1` → a⃑ (combining right harpoon above)
- \tilde (U+0303): `a\u0303` → ã (combining tilde)
- \tvec (U+20E1): `a\u20E1` → a⃡ (combining left right arrow above)
- \vec (U+20D7): `a\u20D7` → a⃗ (combining right arrow above)

**Comment by brichwin** [comment]
Note: The [UnicodeMath 3.1 technical note](https://www.unicode.org/notes/tn28/UTN28-PlainTextMath-v3.1.pdf) on page 28 describes how to create negated operators by simply placing a forward slash `/` in front of the symbol or existing AutoCorrect code.

For example, to create not less than, simply enter `/<`. Here is a larger list:

|Operator|Negated op|Input|
|----|----|----|
|<| ≮ |/<|
|=| ≠ |/=|
|>| ≯ |/>|
|∃| ∄ |/\exists|
|∈| ∉ |/\in|
|∋| ∌ |/\ni|
|∼| ≁ |/\sim|
|≃| ≄ |/\simeq|
|≅| ≇ |/\cong|
|≈| ≉ |/\approx|
|≍| ≭ |/\asymp|
|≡| ≢ |/\equiv|
|≤| ≰ |/\le|
|≥| ≱ |/\ge|
|≶| ≸ |/\lessgtr|
|≷| ≹ |/\gtrless|
|≽| ⋡ |/\succeq|
|≺| ⊀ |/\prec|
|≻| ⊁ |/\succ|
|≼| ⋠ |/\preceq|
|⊂| ⊄ |/\subset|
|⊃| ⊅ |/\superset|
|⊆| ⊈ |/\subseteq|
|⊇| ⊉ |/\superseteq|
|⊑| ⋢ |/\sqsubseteq|
|⊒| ⋣ |/\sqsuperseteq|

**Reply by jenmarsala** [reply]
Thinking about these symbols some more... Even though the forward slash solution is a straightforward one to produce a negated operation, I believe that many people simply wouldn't know to ever try it. I still believe that we may want to ask Microsoft to add alias codes (containing the word "not") for some of the more common negation-related symbols. I also think it would be helpful if Microsoft displays the autocorrect codes containing the forward slash when they are hovered over in the related galleries. So ideally, if I hover over the "not a subset" symbol, I would like to see the commands /\subset and \notsubset.

**Comment by brichwin** [comment]
I added a couple symbols while creating the wiki page that weaves these suggestions together:
|Name | Symbol | Keystrokes to produce the symbol | Recommended autocorrect code(s) | Notes |
| -- | -- | -- | -- | -- |
|Cents sign| ¢ | a2 Alt + x | \cents |Commonly used to help young students understand money-related arithmetic.|
|Divides; is divisible by | &#x2223; | 2223 Alt + x<hr>/mid|\divide<br>\divides|This alias directly describes the meaning of the symbol, making it more intuitive, especially for users who may not be familiar with the existing \mid code. It avoids conflicts with existing codes (like \div for division).|

**Reply by jenmarsala** [reply]
Good additions, Brian! Thank you!

**Comment by brichwin** [comment]
How could we miss the logic symbols? I've added some more suggested short codes for logical symbols. These suggestions could really use a review.

|Name | Symbol | Keystrokes to produce the symbol | Recommended autocorrect code(s) | Notes |
| -- | -- | -- | -- | -- |
|Conjunction (AND), logical AND | &#x2227; | \vee  |\and<br>\land||
|Disjunction (OR), logical OR | &#x2228; | \wedge |\or<br>\lor||
|Not AND (NAND)| &#x22bc; | 22bc Alt + x | \nand | |
|Not OR (NOR)| &#x22bd; | 22bd Alt + x | \nor ||
|Exclusive OR (XOR) | &#x2295; | \oplus | \xor ||
|Exclusive NOR (XNOR)| &#x2299; | \odot | \xnor ||
|Proves| &#x22a2; | \vdash | \proves ||
|Tautology (Always True)| &#x22a4; | \top | \tautology ||
|Contradiction (Always False| &#x22a5; | \bot |\false<br>\contradiction ||
|Implication (If...then)| &#x2192; | \rightarrow | \implication ||
|Biconditional (If and only if)| &#x2194; | \leftrightarrow | \biconditional ||
|Forces| &#x22a9; | 22a9 Alt + x | \forces ||
|Entailment| &#x22a8; | \models | \entailment <br>\true||

**Reply by jenmarsala** [reply]
Good catch, Brian!!! There are some symbols on the list that I've never needed to use; I will leave it to others who use logic symbols more often to review and decide if anything is missing, should be omitted or edited, etc.

**Comment by brichwin** [comment]
I added a "Background and Need" section to the proposed list of new math AutoCorrect short codes. It thought it would be helpful when pressing the case to Microsoft but also as a guiding direction to help in focusing our thought processes for divining and selecting the proposed short codes.

I feel like there is a fine line between providing meaning-based alternatives. In some cases the benefits are obvious (meaning vs shape-based codes as the case of suggesting `\union` for `\cup`). In other cases I think there is a benefit in needing to know that a given symbol is used by convention ( `μ` for the population mean, or a double-struck R (`ℝ`) for the set of Real Numbers). This makes me stop short of suggesting short codes like `\popmean` or `\reals`.  Does this make sense?

As I don't want to speak for the whole group and I am not known for brevity, it would be great if others review that section!

**Reply by jenmarsala** [reply]
I had originally suggested \reals for \doubleR (and similar additions for integers, natural numbers, etc.) but as I think more about this, I believe you're right. I'll comment on the relevant section of <a href="https://github.com/daisy/math-a11y/discussions/25"> Discussion 25 - Let's Generate a List of Aliases for Existing Math AutoCorrect Codes</a>. Let's leave codes like \doubleR for reals, \doubleZ for integers without suggestions \reals or \integers, and let's leave Greek letter codes as-is without suggesting alternate interpretive codes.

**Comment by brichwin** [comment]
Naming Convention Concern:
Some of the suggested new short codes include numbers (e.g.: \90angle). I've noticed that none of the other AutoCorrect codes include digits/numbers. 

Perhaps this is a question for Murray or Microsoft: Should we avoid numbers in short codes?

**Reply by jenmarsala** [reply]
Good question, Brian. If numbers are an issue within this code, we can ask Microsoft to just use the code \rightangle, instead of both \rightangle and \90angle. 
In Discussion Item #25, I had also recommended the use of numbers in autocorrect codes for nth roots, e.g. \5root for fifth root, etc.

**Comment by brichwin** [comment]
I'm wondering about when students are learning to write repeating decimals. For example, when learning how to represent the fraction one third as a decimal number: 
$`\displaystyle \frac{1}{3}=0.\overline{3}`$

Given the example of `\sqrt(x+5)` in UnicodeMath, should we suggest a new `\repeating` short code that works the above could be written `1/3 = 0.\repeating(3)  `?  It might be more intuitive and closer to the mathematical meaning than the existing `\overbar` and `\overline` options.

**Reply by jenmarsala** [reply]
I like this idea, Brian! If Microsoft can implement it, I might suggest both \repeating and \repeat as autocorrect codes.

**Comment by brichwin** [comment]
Does it make sense to propose:
 - `\forsome` as alternative to `\exists`
 - `\foreach` as alternative to `\forall`

**Reply by brichwin** [reply]
Does it make sense to propose:

- `\belongs` or `\memberof` for ∈ (alternative to \in)
- `\contains` for ∋ (more intuitive than \ni)

When reading "x ∈ S" aloud, some say "x belongs to S" or "x is a member of S". 
If I remember right, for "S ∋ x", it's often said as "S contains x as a member" while the existing short code of `\ni` almost implies "not in".

**Reply by jenmarsala** [reply]
I'm fine with all additions proposed here (\forsome, \foreach, \belongs, \memberof, and \contains) as described by Brian.

**Comment by GeorgeKerscher** [comment]
I agree that the current descriptions are meaningless. The question becomes what to recommend as a best practice and develop a canonical  list in US English. This could then be the basis for translations.
